File: 프로그래머스/2/12953. N개의 최소공배수/N개의 최소공배수.java
import java.util.*;

class Solution {
    public int solution(int[] arr) {
        int answer = 0;
        
        
        Arrays.sort(arr);
        for(int i = arr[0] == 1 ? 1 : 2; i <= 100000000; i++) {
            int count = 0;
            for(int j = 0; j < arr.length; j++) {
                if(i % arr[j] == 0) count++;
            }
            if(count == arr.length) {
                answer = i;
                break;
            } 
        }
        
        return answer;
    }
}

File: 프로그래머스/2/12953. N개의 최소공배수/N개의 최소공배수.py
def solution(arr):
    answer = 0

    for i in range(0, len(arr) - 1):
        if i == 0:
            answer = get_lcm(arr[i], arr[i+1])
        else:
            answer = get_lcm(answer, arr[i+1])    

    return answer

def get_lcm(n1, n2) -> int:
    def get_gcd(n1, n2) -> int:
        if n1 % n2 != 0:
            return get_gcd(n2, n1 % n2)
        else:
            return n2

    result = (int)((n1 * n2) / get_gcd(n1, n2))

    return result